vendoring script removes ugorji dependency

`hack/vendor.sh` removes `github.com/ugorji/go/codec` package even when there is a dependency on it from `vendor/src/github.com/coreos/etcd/client/keys.generated.go`. This results in build error : 

```
Building: bundles/1.10.0-dev/binary/docker-1.10.0-dev
vendor/src/github.com/coreos/etcd/client/keys.generated.go:11:2: cannot find package "github.com/ugorji/go/codec" in any of:
	/usr/local/go/src/github.com/ugorji/go/codec (from $GOROOT)
	/go/src/github.com/ugorji/go/codec (from $GOPATH)
	/go/src/github.com/docker/docker/vendor/src/github.com/ugorji/go/codec
make: *** [binary] Error 1
```

I guess everyone who is facing this issue has found a hack-around. 
opening this issue to track it to completion.

--- a/hack/.vendor-helpers.sh
+++ b/hack/.vendor-helpers.sh
@@ -117,10 +117,19 @@ clean() {
 		# This directory contains only .c and .h files which are necessary
 		-path vendor/src/github.com/mattn/go-sqlite3/code
 	)
+
+	# This package is required to build the Etcd client,
+	# but Etcd hard codes a local Godep full path.
+	# FIXME: fix_rewritten_imports fixes this problem in most platforms
+	# but it fails in very small corner cases where it makes the vendor
+	# script to remove this package.
+	# See: https://github.com/docker/docker/issues/19231
+	findArgs+=( -or -path vendor/src/github.com/ugorji/go/codec )
 	for import in "${imports[@]}"; do
 		[ "${#findArgs[@]}" -eq 0 ] || findArgs+=( -or )
 		findArgs+=( -path "vendor/src/$import" )
 	done
+
 	local IFS=$'\n'
 	local prune=( $($find vendor -depth -type d -not '(' "${findArgs[@]}" ')') )
 	unset IFS

